packages/jest-runtime/src/ScriptTransformer.js

```diff
@@ -135,6 +135,13 @@ class ScriptTransformer {
 
       // $FlowFixMe
       transform = (require(transformPath): Transformer);
+
+      if (transform.createTransformer && this._config.transformPlugins) {
```

Comment:
If `transformPlugins` do not exist in config bypass this part; otherwise override the transformer with a new transformer with the passed in plugins.